Changed the way we name downloaded zip files in `build_usd.py` to avoid file name collisions.

Cewein opened the pull request:
### Description of Change(s)

Changed the way we name downloaded zip files in `build_usd.py` to minimise the chance of file collision. Files from a GitHub or GitLab repository previously only had their version numbers as their names. Now, they will incorporate the project name and the version.

I assume other URLs provide this (i.e., Boost and HDF5).

### Fixes Issue(s)
- https://github.com/PixarAnimationStudios/OpenUSD/issues/457

<!--
Please follow the Contributing and Building guidelines to run tests against your
change. Place an X in the box if tests are run and are all tests passing.
-->
- [x] I have verified that all unit tests pass with the proposed changes
<!-- 
Place an X in the box if you have submitted a signed Contributor License Agreement.
A signed CLA must be received before pull requests can be merged.
For instructions, see: http://openusd.org/release/contributing_to_usd.html
-->
- [X] I have submitted a signed Contributor License Agreement

Cewein committed:
Rename files when downloading for safety

Rename files when downloading to minimize the chance of file collision. Files from a GitHub or GitLab repository previously only had their version numbers as their names. Now, they will incorporate the project name and the version.

We assume other URLs provide this (i.e., Boost and HDF5).

nvmkuruc commented on build_scripts/build_usd.py:
```diff
@@ -539,8 +539,15 @@ def DownloadURL(url, context, force, extractDir = None,
+
+        # Change filename for a file form a github/gitlab repository. 
+        # Filename becomes projectName-version to avoid file collision.
+        if ("github" or "gitlab") in url:
```
The expression `"github" or "gitlab"` will always evaluate to `"github"` so it's not actually checking for the presence of `gitlab` in the URL.

The only usage of `gitlab` in `build_usd.py` points to a named release as well, so you can probably just drop the check for `gitlab`.
```
> grep -r gitlab build_scripts/
build_scripts/build_usd.py:TIFF_URL = "https://gitlab.com/libtiff/libtiff/-/archive/v4.0.7/libtiff-v4.0.7.zip"
```

Cewein replied:
Thank you for pointing this out. I checked the GitLab too quickly, since the output name contains the name of the project. I looked into multiple GitLab repositories online, they always come with the project name and the check can be removed.

nvmkuruc commented on build_scripts/build_usd.py:
```diff
@@ -539,8 +539,15 @@ def DownloadURL(url, context, force, extractDir = None,
-        # Extract filename from URL and see if file already exists. 
-        filename = url.split("/")[-1]       
+        # Extract filename from URL.
+        filename = url.split("/")[-1]
```
I'd consider if it would be better to use `urllib.parse.urlparse` to breakdown the `url` into its components (https://docs.python.org/3/library/urllib.parse.html).  That should be more robust than splitting on `/`.

It might also make things more robust to just use the full `path` component of the URL when creating a target to download the files rather than trying to decompose based on the naming conventions of particular repositories.  I'm not sure if there's any drawbacks to that.

Cewein replied:
The only drawback that I can see with the name as the path is if the filename become too long, it might not save properly.

Using `urllib.parse.urlparse` is a good idea though.

Cewein committed:
added urlparse and better if statement

The URL is now split into its components using `urlparse`. We check if the network location of the URL is github and change the name accordingly.

jesschimein commented:
Filed as internal issue #USD-9603

:heavy_exclamation_mark: Please make sure that a signed CLA has been submitted!

jesschimein commented:
/AzurePipelines run
